index.js

```diff
@@ -116,9 +117,9 @@ class KeyringController extends EventEmitter {
       throw new Error('Seed phrase is invalid.');
     }
 
-    this.clearKeyrings();
+    this.password = password;
 
-    await this.persistAllKeyrings(password);
```

Comment:
I'm guessing this call was primarily meant to set the password, so it has been removed.

It did have another functional impact, in that it ensured the `clearKeyrings` operation was reflected in the vault, in case the attempt to create a new vault fails. But... that situation is not anticipated. At this point we have validated that the SRP is valid, and the password is text, so there aren't any more additional failure conditions.

Moreover; leaving an existing vault in state isn't an awful end result to encountering an unexpected error. If anything it might aid with recovery, if we do somehow end up accidentally bricking the wallet with an update.